src/types/process.go

```diff
@@ -15,7 +15,8 @@ type ProcessConfig struct {
 	Name              string
 	Disabled          bool                   `yaml:"disabled,omitempty"`
 	IsDaemon          bool                   `yaml:"is_daemon,omitempty"`
-	Command           string                 `yaml:"command"`
+	Command           *string                `yaml:"command,omitempty"`
+	Entrypoint        *[]string              `yaml:"entrypoint,omitempty"`
```

Comment:
Why `entrypoint` is a pointer?

Reply:
since when Entrypont is `!= nil` we disregard `ShellConfig`

https://github.com/F1bonacc1/process-compose/blob/b99cd40adbde9ff32c6aebae0f357671fd091ce8/src/types/validators.go#L98-L107

Reply:
Yes, I saw that.
Can the same result be achieved if you check that it's not empty?

Reply:
The point is that you can set it to empty to execute a command directly, without shell, if the command is simply a name or path of an executable.  

Current implementation allows one to:

1. Use command on it’s own, executed via shell set in ShellConfig. This is the old behaviour
2. Use Entrypoint on it’s own
3. Use Entrypoint with command

In case 2 and 3, ShellConfig is ignored. If Entrypoint happens to be an empty array, the command becomes the executable, since the command (if not nil) is simply appended to the Entrypoint array.

If you think the latter case is not useful, we could indeed default Entrypoint to [] and have it result in behaviour number 1

Reply:
I think that cases 1 and 2 are very useful. 3 however, is a bit confusing and doesn't add anything to the functionality that can't be achieved by 1 and 2.
IMHO the flow can be simplified to the following rules:

1. If `command` is not empty (`len(command) > 0`) - run the command. This is the default behavior.
2. If `entrypoint` is not empty - run the `entrypoint` exec `[0]` and args `[1:]`.
3. If both are not empty:
   a. issue an error to log during validation
   b. run the `command`
   c. ignore the `entrypoint`
   d. in strict mode (upcoming) fail the start

What do you think?

Reply:
Tbh, I don't personally like this pointer business (would very much rather use an Option type, which `go` doesn't seem to provide) so would be happy to use `[]` and `""` signifying "not set" for `Entrypoint` and `command` respectively, if you are happy to drop the use case described above

Reply:
I wrote my reply without seeing yours just above :)

Reply:
> 1. If `command` is not empty (`len(command) > 0`) - run the command. This is the default behavior.
> 2. If `entrypoint` is not empty - run the `entrypoint` exec `[0]` and args `[1:]`.
> 3. If both are not empty:
>    a. issue an error to log during validation
>    b. run the `command`
>    c. ignore the `entrypoint`
>    d. in strict mode (upcoming) fail the start
> 

Sounds good. 

Btw. I just refreshed my memory regarding why I implemented it to support `Entrypoint` + `command`: this is how docker and docker-compose work (they also allow both of them to be either a single string or an array). Tbh I always found it to be quite confusing so it's probably not the best model to replicate.